### ISSUE TITLE
Converter produces empty SPC song

### ISSUE THREAD

**Issue by masonwheeler** [alekmaul/pvsneslib#61]
I'm attempting to convert an IT file.  It has 8 channels, 8 patterns, 16 instruments and 30 samples.  Should be well within operational tolerances.  The converter produces a SPC file that plays for the right length of time, but no sound comes out; it's silence on all 8 channels.

Is there any way to figure out what's going wrong and how to address it?

**Comment by alekmaul** [comment]
Yeah, if your samples are not OK or if you are using some effects not supported by SNESMOD, you will have silence.
Take a look here : https://github.com/alekmaul/pvsneslib/blob/master/pvsneslib/pvsneslib_snesmod.txt

**Comment by masonwheeler** [comment]
> You also need to have one instrument per samples, it is the case ?

No, this is most likely the issue.  I wasn't aware that was a limitation.

If these are known issues that will cause conversion to fail, could the code check for them and return an error explaining the problem if they're found, rather than producing (quite literally!) silent failure?

**Comment by hoit** [comment]
i agree with you, it will be better if smconv tool (to convert .it file) detect this kind of things. But the tool was not developed for pvsneslib like some others and there are a lot of things to do on pvsneslib itself.
I do not know if Alekmaul will add it to the roadmap but If you want to treat it, it will be usefull for everybody :)

**Comment by masonwheeler** [comment]
I might have a look if I have time, but I'm not a C++ coder and this code, as currently written, won't even compile on Visual Studio.  Looks like it's built for some other compiler that has slightly different ideas about what constitutes acceptable C++ code.

**Comment by masonwheeler** [comment]
@alekmaul 
> I use Alt-E in Schism Tracker to perform the resize sample command.

I'm confused by this part.  Just downloaded Schism Tracker, and that command doesn't appear to exist...

**Comment by masonwheeler** [comment]
All right, after spending several hours bringing it down to 1 sample per instrument, with all samples having length and loop offsets divisible by 16... it's still producing a silent SPC file.

I'm totally lost at this point.

**Comment by alekmaul** [comment]
could you please post the source code you use for music playing ?

**Comment by hoit** [comment]
Does this [ticket](https://github.com/alekmaul/pvsneslib/issues/48) can help you?

**Comment by masonwheeler** [comment]
> could you please post the source code you use for music playing ?

```
	// Initialize sound engine (take some time)
	spcBoot();
	
    // Initialize SNES 
	consoleInit();
    
	// Set give soundbank
	spcSetBank(&SOUNDBANK__);
	
	// allocate around 10K of sound ram (39 256-byte blocks)
	spcAllocateSoundRegion(39);

	// Load music
	spcLoad(MOD_BGM);
	
	// Play file from the beginning
	spcPlay(0);
	spcSetModuleVolume(100);
```

Pretty standard copy/paste code. The problem isn't in the game code; it's in the SPC file.  I've verified with multiple different SPC players that the result of conversion is silent.

**Comment by KungFuFurby** [comment]
Ah yes, some of my notes from years ago are here.

The very first thing on my mind is... better make sure that, going by Schism Tracker, "Control" is set to "Instruments" in the .it file. That's probably the most common cause of things going silent that I can think of... or at least the playback going bad.

**Comment by alekmaul** [comment]
also, during your loop of music playing, do you use spcProcess ?
